File: backend/src/main/java/io/github/matheusvdlima/incidents/entity/Usuario.java
package io.github.matheusvdlima.incidents.entity;

import io.github.matheusvdlima.incidents.enums.PerfilEnum;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "tb_usuario")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 150)
    private String nome;

    @Column(nullable = false, length = 120, unique = true)
    private String email;

    @Column(nullable = false)
    private String senha;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    private PerfilEnum perfil;

    @Column(name = "data_criacao", updatable = false)
    @CreationTimestamp
    private LocalDateTime dataCriacao;

}

File: frontend/src/app/models/usuario.ts
import { Dominio } from "./dominio";

export interface Usuario {
  id?: string;
  nome: string;
  email: string;
  senha: string;
  perfil: Dominio;
  dataCriacao: string;
}

export interface UsuarioParametro {
  page: number,
  size: number,
  sort?: string,
  texto?: string,
  perfil?: string
}
